splunk-quarkus/src/main/java/com/redhat/console/integrations/splunk/SplunkIntegration.java

```diff
@@ -114,6 +114,8 @@ private void configureHandler() throws Exception {
                 // aggregate transformed messages and append them together
                 // aggregate by "metadata" header as it contains data unique per target splunk instance
                 .aggregate(header("metadata"), new EventAppender())
+                // use default of 10 threads to process aggregated records (that act as queue)
+                .parallelProcessing(true)
```

Comment:
This should be tread-safe by the strategy key which is the metadata header expression. I think we should test this out.

Aggregation would use a thread pool. The fact that the concurrency is added here is to provide of concurrent processing of integrations (concurrent HTTP calls), as the aggregation bucket act as a queue.

Reply:
I think I misunderstood the code because of my poor Camel knowledge 😄 My first impression was that `parallelProcessing` only affected the aggregation of the `events`, which is basically a `String` concatenation as implemented in `EventAppender`. I guess `parallelProcessing` has an impact on everything that follows that instruction.

Reply:
I had a similar understanding before, however the reality is different and it indeed affects other instructions.